### PR TITLE
[KYUUBI #3051] Support to get the  real client ip address for thrift connection when using VIP as kyuubi server load balancer

### LINKED ISSUE #3051: [FEATURE] Support to get real ip for thrift connection when using VIP as kyuubi server LB
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

We deploy kyuubi server on k8s and using k8s lb and the connection path is:

client -> lb -> kyuubi server.

Now the client ip address is always the lb ip and is meaning less.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Close #3051 

We deploy kyuubi server on k8s and using k8s lb and the connection path is:

client -> lb -> kyuubi server.

Now the client ip address is always the lb ip and is meaning less.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiReservedKeys.scala [hunk 1]
```diff
@@ -19,6 +19,7 @@ package org.apache.kyuubi.config
 
 object KyuubiReservedKeys {
   final val KYUUBI_SESSION_USER_KEY = "kyuubi.session.user"
+  final val KYUUBI_SESSION_IP_KEY = "kyuubi.session.ip"
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiReservedKeys.scala:22]: we have `kyuubi.client.ip` already, how about using this?

maybe  kyuubi.client.ipAddress is better

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiReservedKeys.scala:22]: thanks for the information

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/package.scala [hunk 2]
```diff
@@ -28,5 +28,5 @@ package object session {
 
   val SPARK_PREFIX = "spark."
 
-  val CLIENT_IP_KEY = "kyuubi.client.ip"
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/package.scala:31]: move it to reserved keys class

**Comment by yaooqinn** [comment]
for authentication, this introduce a CVE

**Comment by turboFei** [comment]
> for authentication, this introduces a CVE

Seems difficult to avoid it....

**Comment by turboFei** [comment]
In fact, now with the proxy lb, the client ip address is also not the real ip and maybe not in the ip white list.

**Comment by yaooqinn** [comment]
> In fact, now with the proxy lb, the client ip address is also not the real ip and maybe not in the ip white list.

that’s lb's duty..

**Comment by turboFei** [comment]
make sense

**Comment by yaooqinn** [comment]
if we just want to track the IP, not for authentication, we may just leave it in the configuration and get it whenever we want

**Comment by turboFei** [comment]
> if we just want to track the IP, not for authentication, we may just leave it in the configuration and get it whenever we want

How about that,
we define another reserved key for this?

Otherwise, the kyuubi.client.ip will be overwrite.

**Comment by turboFei** [comment]
It will be overwritten here:

https://github.com/apache/incubator-kyuubi/blob/99934591f07be5bc8abc1955848e4fc291b8569b/kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala#L60-L65

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thank a lot @yaooqinn

File: externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionImpl.scala [hunk 3]
```diff
@@ -33,7 +33,6 @@ class HiveSessionImpl(
     protocol: TProtocolVersion,
     user: String,
     password: String,
-    serverIpAddress: String,
```
Comment (pan3793) [inline externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionImpl.scala:36]: this change may breaks https://github.com/apache/incubator-kyuubi/pull/1530/files#diff-759a09fe90c07b0ef78f6f4dc28bac9e3d0785d4b922cbe33422524e8a2520d4R51

Comment (pan3793) [reply externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionImpl.scala:36]: cc @turboFei @wForget

Comment (wForget) [reply externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionImpl.scala:36]: [SparkSessionImpl.scala#L51](https://github.com/turboFei/incubator-kyuubi/blob/97f9b00417f211cc547998821ba421b15edb6c76/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala#L51) becomes redundant, thanks, I will remove it.

```
  def serverIpAddress(): String = serverIpAddress
```

**Pull request closed without merging**